compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.td

```diff
@@ -333,11 +333,17 @@ def IREEGPU_TargetWgpAttr : AttrDef<IREEGPU_Dialect, "TargetWgp"> {
     // The maximal number of threads per X/Y/Z dimension in one workgroup.
     "DenseI32ArrayAttr":$max_workgroup_sizes,
     // The maximal number of threads we can have in one workgroup.
-    "uint32_t":$max_thread_count_per_workgroup,
+    "int32_t":$max_thread_count_per_workgroup,
     // The maximal number of shared memory bytes we can allocate per workgroup.
-    "uint32_t":$max_workgroup_memory_bytes,
-    // Tthe maximum number of workgroups per X/Y/Z dimension in a dispatch.
+    "int32_t":$max_workgroup_memory_bytes,
+    // The maximum number of workgroups per X/Y/Z dimension in a dispatch.
     "DenseI32ArrayAttr":$max_workgroup_counts,
+    // Max load instruction size in bits
+    OptionalParameter<"std::optional<int32_t>">:$max_load_instruction_bits,
+    // Number of SIMDs per workgroup
+    OptionalParameter<"std::optional<int32_t>">:$workgroup_simds,
+    // VGPR register space size in bits
+    OptionalParameter<"std::optional<int32_t>">:$vgpr_space_bits,
```

Comment:
I followed the discussion on discord about why they are optional, very helpful. Can we add a comment that explains why they are optional in the td file?

Reply:
Filed https://github.com/iree-org/iree/issues/18849 , will reference it here in a TODO comment.